### PR TITLE
Added chat functionalities

### PR DESCRIPTION
**Pull request by Paukky**
+ Renamed Authcontext.tsx to FirebaseContext

### CODE REVIEW

**Review by kouglas — APPROVED** [review]
works well so far. great work. the login and logout buttons will likely be moved in the final iteration.

File: src/components/ChatItemBox.tsx [hunk 1]
```diff
@@ -8,15 +8,17 @@ import React from "react";
 import ChatItem from "../types/ChatItem";
 
 interface ChatItemProps {
-	username: String;
+	name: String;
 	message: String;
-	timestamp: Date; // optional
+	timestamp: number; // optional
 }
```
Comment (Z-Index-1) [inline src/components/ChatItemBox.tsx:14]: Glad to see the typing changes

File: src/components/Header.tsx [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+		{currentUser
+			? <React.Fragment>
+					<p>Hello, {currentUser.displayName}</p>
+					<button onClick={logOut}>Log out</button>
+				</React.Fragment>
+			: 
+			<button onClick={signInWithGoogle}>Sign in with Google</button>
+		}
+	</div>
+
```
Comment (Z-Index-1) [inline src/components/Header.tsx:18]: This will need to be moved to the login page, in a new PR

Edit: Correction, it's already been worked on

**Comment by Z-Index-1** [comment]
Atleast from a technical level, this works on my end. I'd have liked to see some error handling on the promises which return void though, but if this is what's documented that's understandable. On the aesthetics and UI/UX side however, there are quite a few issues. The text is white on my end, the "log in" button on chrome is blacked out, and the proportions are wrong. These issues were not the main concern of this PR however ,but they should be noted. 

All in all though, fantastic work

**Review by Z-Index-1 — APPROVED** [review]

**Pull request merged**